base/options.go

```diff
@@ -1,19 +1,52 @@
 package base
 
 import (
+	"flag"
+	"fmt"
+	"os"
 	"os/exec"
 
 	"github.com/keybase/go-keybase-chat-bot/kbchat"
 )
 
+type AWSOptions struct {
+	AWSRegion          string
+	CloudWatchLogGroup string
+}
+
 type Options struct {
 	KeybaseLocation string
 	Home            string
 	Announcement    string
 	DSN             string
+	AWSOpts         *AWSOptions
+}
+
+func (o *Options) Parse(fs *flag.FlagSet, argv []string) error {
+	if len(argv) <= 1 {
+		return fmt.Errorf("Bad usage: no arguments specified")
+	}
+	fs.StringVar(&o.KeybaseLocation, "keybase", "keybase", "keybase command")
+	fs.StringVar(&o.Home, "home", "", "Home directory")
+	fs.StringVar(&o.Announcement, "announcement", os.Getenv("BOT_ANNOUNCEMENT"),
+		"Where to announce we are running")
+	fs.StringVar(&o.DSN, "dsn", os.Getenv("BOT_DSN"), "Bot database DSN")
+	if o.AWSOpts == nil {
+		o.AWSOpts = &AWSOptions{}
+	}
+	fs.StringVar(&o.AWSOpts.AWSRegion, "aws-region", os.Getenv("BOT_DSN"), "AWS region for cloudwatch logs, optional")
+	fs.StringVar(&o.AWSOpts.CloudWatchLogGroup, "cloudwatch-log-group", os.Getenv("BOT_CLOUDWATCH_LOG_GROUP"), "Cloudwatch log group name, optional")
+	if err := fs.Parse(argv[1:]); err != nil {
+		return err
+	}
+	if len(o.DSN) == 0 {
```

Comment:
Every bot needs this? Doesn't seem right.

Reply:
mmm true